python/IAM_POLICY_REQUIRED/IAM_POLICY_REQUIRED.py

```diff
@@ -0,0 +1,526 @@
+            for policy in attached_policies:
+                managed_policies.append(policy['PolicyArn'])
+                if list_contains_all(managed_policies, policy_arns):
+                    return True
+
+        return False
+    elif resource_type == 'AWS::IAM::Role':
+        return list_contains_all(get_attached_policies(configuration_item), policy_arns)
+    else:
+        raise ValueError('Unable to handle resource type {}'.format(resource_type))
+
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+    """Form the evaluation(s) to be return to Config Rules
+
+    Return either:
+    None -- when no result needs to be displayed
+    a string -- either COMPLIANT, NON_COMPLIANT or NOT_APPLICABLE
+    a dictionary -- the evaluation dictionary, usually built by build_evaluation_from_config_item()
+    a list of dictionary -- a list of evaluation dictionary , usually built by build_evaluation()
+
+    Keyword arguments:
+    event -- the event variable given in the lambda handler
+    configuration_item -- the configurationItem dictionary in the invokingEvent
+    valid_rule_parameters -- the output of the evaluate_parameters() representing validated parameters of the Config Rule
+
+    Advanced Notes:
+    1 -- if a resource is deleted and generate a configuration change with ResourceDeleted status, the Boilerplate code will put a NOT_APPLICABLE on this resource automatically.
+    2 -- if a None or a list of dictionary is returned, the old evaluation(s) which are not returned in the new evaluation list are returned as NOT_APPLICABLE by the Boilerplate code
+    3 -- if None or an empty string, list or dict is returned, the Boilerplate code will put a "shadow" evaluation to feedback that the evaluation took place properly
+    """
+    policy_arns = valid_rule_parameters['policyArns']
+    exception_list = valid_rule_parameters["exceptionList"]
+    ignored_roles = exception_list["roles"]
+    ignored_users = exception_list["users"]
+
+    if should_ignore_config_item(configuration_item, ignored_roles, ignored_users):
+        return build_evaluation_from_config_item(configuration_item, 'COMPLIANT', 'Ignored IAM entity')
+    elif has_policy_attached(event, configuration_item, policy_arns):
+        return build_evaluation_from_config_item(configuration_item, 'COMPLIANT', 'All expected policies attached')
+    else:
+        return build_evaluation_from_config_item(configuration_item, 'NON_COMPLIANT', 'IAM entity missing policies')
+
+
+def is_valid_arn(arn):
+    pattern = re.compile("arn:(aws[a-zA-Z-]*)?:iam::(aws|\d{12}):policy\/[a-zA-Z0-9-_\/]+")
+    return pattern.match(arn)
+
+
+def extract_entities_from_exception_list(entity_type, exception_list):
+    pattern = re.compile("{Type}:\s?\[([a-zA-Z0-9-_,]+)\]".format(Type=entity_type))
+    matches = pattern.search(exception_list)
+    if matches:
+        return matches.group(1).replace(' ', '').split(",")
+    else:

```

Comment:
Remove this `else:`, as the if statement include a return.

Reply:
updated